File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0) return 0;
        int longest = 1 , count = 0 , lastsmall = INT_MIN;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size();i++){
            if(nums[i]-1 == lastsmall){
                count++;
                lastsmall = nums[i];
            }else if(nums[i] != lastsmall){
                count = 1;
                lastsmall = nums[i];
            }
            longest = max(longest,count);
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        SortedSet<Integer> numSet = new TreeSet<>();
        for (int num : nums) {
            numSet.add(num); // TreeSet automatically sorts elements and removes duplicates
        }

        int longest = 1, count = 0, lastSmall = Integer.MIN_VALUE;
        for (int num : numSet) {
            if (num - 1 == lastSmall) {
                count++;
                lastSmall = num;
            } else {
                count = 1;
                lastSmall = num;
            }
            longest = Math.max(longest, count);
        }
        return longest;
    }
}